storage/buckets/main.go

```diff
@@ -208,15 +213,32 @@ func removeUser(c *storage.Client, bucketName string) error {
 	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 	defer cancel()
 	bucket := c.Bucket(bucketName)
-	policy, err := bucket.IAM().Policy(ctx)
+	policy, err := bucket.IAM().V3().Policy(ctx)
 	if err != nil {
 		return err
 	}
 	// Other valid prefixes are "serviceAccount:", "user:"
 	// See the documentation for more values.
 	// https://cloud.google.com/storage/docs/access-control/iam
-	policy.Remove("group:cloud-logs@google.com", "roles/storage.objectViewer")
-	if err := bucket.IAM().SetPolicy(ctx, policy); err != nil {
+	for _, binding := range policy.Bindings {
+		// Only remove unconditional bindings matching role
+		if binding.Role == "roles/storage.objectViewer" && binding.Condition == nil {
+			// Filter out member.
+			i := -1
+			for j, member := range binding.Members {
+				if member == "group:cloud-logs@google.com" {
+					i = j
```

Comment:
Can there be more than one instance with this as the member? I think that was part of the more complex logic before.

Reply:
There can only be one binding matching role with no conditions. We use the same logic in other samples as well: https://github.com/googleapis/nodejs-storage/blob/master/samples/removeBucketIamMember.js#L46